staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1070,31 +1066,33 @@ func (vs *VSphere) DisksAreAttached(nodeVolumes map[k8stypes.NodeName][]string)
 				VC_DC := nodeInfo.vcServer + nodeInfo.dataCenter.String()
 				dcNodes[VC_DC] = append(dcNodes[VC_DC], nodeName)
 			}
-
+		
+			rNodes := make(chan []k8stypes.NodeName)
+			ech := make(chan error)
 			for _, nodeNames := range dcNodes {
 				localAttachedMap := make(map[string]map[string]bool)
 				localAttachedMaps = append(localAttachedMaps, localAttachedMap)
 				// Start go routines per VC-DC to check disks are attached
-				go func(nodes []k8stypes.NodeName) {
+				go func(nodes []k8stypes.NodeName, rNodes chan []k8stypes.NodeName, ech chan error) {
 					nodesToRetryLocal, err := vs.checkDiskAttached(ctx, nodes, nodeVolumes, localAttachedMap, retry)
 					if err != nil {
 						if !vclib.IsManagedObjectNotFoundError(err) {
-							globalErrMutex.Lock()
-							globalErr = err
-							globalErrMutex.Unlock()
 							klog.Errorf("Failed to check disk attached for nodes: %+v. err: %+v", nodes, err)
+							ech <- err
 						}
 					}
-					nodesToRetryMutex.Lock()
-					nodesToRetry = append(nodesToRetry, nodesToRetryLocal...)
-					nodesToRetryMutex.Unlock()
-					wg.Done()
-				}(nodeNames)
-				wg.Add(1)
+					rNodes <- nodesToRetryLocal
+				}(nodeNames, rNodes, ech)
 			}
-			wg.Wait()
-			if globalErr != nil {
-				return nodesToRetry, globalErr
+
+			// Collect the retry nodes/errors list from goroutines asynchronously
+			for retryCount := dcNodes {
```

Comment:
Is this supposed to be `range dcNodes`?

Reply:
Yes, sorry that's a typo. Thanks for catching.